content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+  It's a good practice to match P&T Composition's behavior by setting
+  `metadata.generateName: "name-of-the-xr-"` for any new desired resources.
+* Don't blindly add new entries to the desired resources array every time your
+  function is invoked. Remember to check whether your desired resource is
+  already in the `observed` and/or `desired` objects. You may need to update it
+  rather than create it.
+* Don't bypass providers. Composition Functions are designed to tell Crossplane
+  how to orchestrate managed resources - not to directly orchestrate external
+  systems.
+* Include your function name and version in any results you return to aid in
+  debugging.
+* Write tests for your function! Pass it a `FunctionIO` on stdin in and ensure
+  it returns the expected `FunctionIO` on stdout.
+* Keep your Functions fast and lightweight. Remember that Crossplane will run
+  them approximately once every 30-60 seconds.
+
+## The xfn Runner
+
+Composition Function runners are designed to be pluggable. Each time Crossplane
+needs to invoke a Composition Function it makes a gRPC call to a configurable
+endpoint. The default, reference Composition Function runner is named `xfn`.
+
+{{< hint "note" >}}
+The default runner endpoint is `unix-abstract:crossplane/fn/default.sock`. It's
+possible to run Functions using a different runner, for example:
+
+```yaml
+  functions:
+  - name: my-cool-Function
+    type: Container
+    container:
+      image: xkpg.io/my-cool-Function:0.1.0
+      runner:
+        endpoint: unix-abstract:/your/custom/runner.sock
+```
+
+Currently Crossplane uses unauthenticated, unencrypted gRPC requests to run
+Functions, so requests should not be sent over the network. Encryption and
+authentication will be added in a future release.
+{{< /hint >}}
+
+`xfn` is deployed as a sidecar container within the Crossplane pod. It runs each
+Composition Function as a nested [rootless container][rootless-containers].
+
+{{< img src="composition-functions-xfn-runner.png" alt="Crossplane running Functions using xfn via gRPC" size="tiny" >}}
+
+The Crossplane Helm chart deploys `xfn` with slightly elevated permissions.
+Namely:
+
+* The [Unconfined seccomp profile][kubernetes-seccomp].
+* `CAP_SETUID` and `CAP_SETGID`.
+
+`xfn` doesn't strictly need either of these. It's possible to run it under
+seccomp but doing so is onerous. You must author a custom profile that allows a
+few additional syscalls required to run containers, including `unshare`,
+`mount`, and `unmount`. Refer to the [seccomp documentation][kubernetes-seccomp]
+for information on how to do so.
+
+Granting `CAP_SETUID` and `CAP_SETGID` is optional. Doing so allows `xfn` to
+create slightly better rootless containers. If `xfn` is not granted these
+capabilities it will not support multiple users and groups inside Composition
+Function containers; only UID and GID 0 will exist. Regardless of capabilities
```

Comment:
Can we make this more precise and straight-forward? For example:

> Granting `CAP_SETUID` and `CAP_SETGID` allows `xfn` to support multiple users
> and groups inside Composition Function containers. Without this setting `xfn`
> uses UID and GID 0.

(I don't know if this edit is technically accurate) 

Write in the affirmative "it does" instead of the negative "it won't support".